### PR TITLE
Consolidate temporary pins for example projects

### PR DESCRIPTION
**Pull request by jmsanders**
Suggested in:

https://github.com/dagster-io/dagster/pull/14811#pullrequestreview-1481827412

This gives us a common file with a brief explanation where we can introduce temporary pins to keep our example tests that install from PyPI working until the next version of Dagster is released.

### CODE REVIEW

**Comment by jmsanders** [comment]
Per our discussion, I also opened this GitHub discussion to track these kinds of suggested pins. https://github.com/dagster-io/dagster/discussions/14817

File: examples/user_pins.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# typically don't patch previously released versions of Dagster to
+# introduce a pin. Instead, introduce a pin or a backward compatible fix
+# in the next release of Dagster.
+
+# We run automated tests to install the latest version Dagster from PyPI
+# and verify that these example projects work. We use this file to
+# temporarily introduce pins for these tests until the next version of
+# Dagster is published to PyPI.
+
+croniter<1.4.0
```
Comment (alangenfeld) [inline examples/user_pins.txt:14]: can already stack remove this

**Review by alangenfeld — APPROVED** [review]
i like this

**Comment by alangenfeld** [comment]
`user_pins.txt` is sort of odd name i think, maybe `workaround_pins`, `temp_pins` or something?

**Pull request merged**